uxarray/grid/coordinates.py

```diff
@@ -319,6 +320,91 @@ def _construct_xyz_centroids(node_x, node_y, node_z, face_nodes, n_nodes_per_fac
     return centroids[0, :], centroids[1, :], centroids[2, :]
 
 
+def _populate_edge_centroids(grid, repopulate=False):
+    """Finds the centroids using cartesian averaging of the edges based off the
+    vertices. The centroid is defined as the average of the x, y, z
+    coordinates, normalized.
+
+    Parameters
+    ----------
+    repopulate : bool, optional
+        Bool used to turn on/off repopulating the edge coordinates of the centroids
+    """
+
+    node_x = grid.node_x.values
+    node_y = grid.node_y.values
+    node_z = grid.node_z.values
+    edge_nodes_con = grid.edge_node_connectivity.values
+
+    if "edge_lon" not in grid._ds or repopulate:
+        # Construct the centroids if there are none stored
+        if "edge_x" not in grid._ds:
+            centroid_x, centroid_y, centroid_z = _construct_edge_centroids(
+                node_x, node_y, node_z, edge_nodes_con
+            )
+
+        else:
+            # If there are cartesian centroids already use those instead
+            centroid_x, centroid_y, centroid_z = grid.edge_x, grid.edge_y, grid.edge_z
+
+        # Convert from xyz to latlon
+        centroid_lon, centroid_lat = _get_lonlat_from_xyz(
+            centroid_x, centroid_y, centroid_z
+        )
+    else:
+        # Convert to xyz if there are latlon centroids already stored
+        centroid_lon, centroid_lat = grid.edge_lon.values, grid.edge_lat.values
+        centroid_x, centroid_y, centroid_z = _get_xyz_from_lonlat(
+            centroid_lon, centroid_lat
+        )
+
+    # Populate the centroids
+    if "edge_lon" not in grid._ds or repopulate:
+        grid._ds["edge_lon"] = xr.DataArray(
+            centroid_lon, dims=["n_edge"], attrs={"units": "degrees_east"}
```

Comment:
This section is taken from #628, which is still a WIP, but please populate the "long name" and "standard name" attributes for all the parts here. 

https://github.com/UXARRAY/uxarray/blob/93b8b947e8d5b5442675d8a1c6cebe9d2bc87efc/uxarray/conventions/ugrid.py#L43-L53

Reply:
For the cartesian ones, I see the long name is "TODO", should I leave it like that or label it something such as "x coordinate of the center of each edge"?

Reply:
Yes that works!